Code/Framework/AzToolsFramework/AzToolsFramework/Metadata/UuidUtils.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace AzToolsFramework
+{
+    struct IUuidUtil
```

Comment:
first time looking at this so pardon my ignorance - why make a separate interace + component for this class instead of adding these functions to the existing `IMetadataRequests` interface and `MetadataManager`?

Reply:
IMetadataRequests is meant to handle metadata files in a generic way, it isn't supposed to have any UUID-specific function (it can be used to store any kind of data meant to be associated with an asset).  UuidUtil is meant to build on top of the metadata system to provide an API for assigning UUIDs to an asset.